### PR TITLE
feat(insights-for-employees): component added and translations

### PR DESCRIPTION
**Pull request by clickspider**
# Changes

#560 

## Added

- Reusable component "date-reports.vue"
- Helper function to getMonthName
- Page for yearly reports
- Button in the header that will lead to the yearly report (where you can also see the page yearly title)

## Changed

- You can now see at the page title the month which you are viewing.
- The /month route is now using the new "date-reports.vue" cmp.

## How to test

1. Login
2. At the header you will see a button called "Yearly overview", that should show you your yearly report, you can go to any year you want. 

## Screenshots

image

### CODE REVIEW

**Comment by winato** [comment]
Can you sync the branch with develop, because the problem with lenra is already solved? 🙂

File: components/reports/date-navigation-buttons.vue [hunk 1]
```diff
@@ -33,22 +34,22 @@
...
-        :title="$t('nextHint')"
+        :title="$t('nextHint', {DATE_MSG})"
         @click="handleNextClick()"
       >
         <b-icon icon="arrow-right" />
       </b-button>
 
       <h2 class="navigation-buttons__week-label--reports text-capitalize">
-        {{$d(selectedDate, 'monthYear')}}
+        {{isYearly ? selectedDate.getFullYear() : $d(selectedDate, 'monthYear')}}
```
Comment (winato) [inline components/reports/date-navigation-buttons.vue:44]: Can it be better to move `selectedDate.getFullYear()` to computed and don't call the method in a template?

Comment (clickspider) [reply components/reports/date-navigation-buttons.vue:44]: Does it really matter? 🗡️

Comment (winato) [reply components/reports/date-navigation-buttons.vue:44]: From a performance perspective - yes. 
The method invocation will always run the function whenever a re-render happens. I think this operation should be cached with computed. And I suppose it is a [classic example](https://vuejs.org/guide/essentials/computed.html#computed-caching-vs-methods) where we should use `computed`

**Review by sanderdejong88 — APPROVED** [review]

**Pull request merged**